nav2_dynamic_params/include/nav2_dynamic_params/dynamic_params_client.hpp

```diff
@@ -100,53 +100,64 @@ class DynamicParamsClient
       {rclcpp::ParameterEventsFilter::EventType::NEW,
         rclcpp::ParameterEventsFilter::EventType::CHANGED});
     if (!(filter.get_events()).empty()) {
-      auto param_msg = ((filter.get_events()).front()).second;
-      auto param_value = rclcpp::Parameter::from_parameter_msg(*param_msg);
-      new_value = param_value.get_value<T>();
-      dynamic_param_map_[param_name] = param_value;
+      new_value = get_param_from_event<T>(filter);
       return true;
     } else {
-      if (dynamic_param_map_.count(param_name) > 0 &&
-        !dynamic_param_map_[param_name].get_type() == rclcpp::ParameterType::PARAMETER_NOT_SET)
-      {
-        new_value = dynamic_param_map_[param_name].get_value<T>();
+      if (get_param_from_map<T>(param_name, new_value)) {
+        return true;
       } else {
-        RCLCPP_DEBUG(rclcpp::get_logger("dynamic_params_client"),
-          "Parameter '%s' not set", param_name.c_str());
+        RCLCPP_WARN(rclcpp::get_logger("dynamic_params_client"),
+          "Parameter '%s' not set on node", param_name.c_str());
+        return false;
       }
-      return false;
     }
   }
 
   template<class T>
-  bool get_event_param(
-    const rcl_interfaces::msg::ParameterEvent::SharedPtr event,
+  bool get_event_param_or(
+    const rcl_interfaces::msg::ParameterEvent::SharedPtr & event,
     const std::string & param_name, T & new_value, const T & default_value)
   {
-    rclcpp::ParameterEventsFilter filter(event, {param_name},
+    if (get_event_param<T>(event, param_name, new_value)) {
+      return true;
+    } else {
+      new_value = default_value;
+      return false;
+    }
+  }
+
+  bool check_in_event(
```

Comment:
Sounds like you want to checkin the event to git or something. A more standard name would be something like
`is_in_event()`. Then the if statement reads like
if(is_in_event())